src/search/components/query-editor.controller.js

```diff
@@ -193,6 +185,15 @@ function QueryEditorController(
       return node.nodes.length > 0;
     });
 
+    if (groupedTree.nodes[0].operator === 'AND' &&
+      groupedTree.nodes.length > 1 &&
+      groupedTree.nodes[0].nodes.length === 1
+    ) {
+      var firstGroup = groupedTree.nodes[0];
+      groupedTree.nodes.splice(0, 1);
+      groupedTree.nodes[0].nodes.push(firstGroup);
```

Comment:
```suggestion
      groupedTree.nodes.splice(0, 1, firstGroup);
```

If you want to replace the element at 0th index, you can also do it like this with splice

Reply:
It's not a replacement. 
I want to add it to the subnodes which are a level lower.